frontend/components/bridge/DepositForBridging.tsx

```diff
@@ -245,47 +227,68 @@ export const DepositForBridging = ({
           );
         }
 
+        const areFundsReceived = totalRequiredInWei - currentBalanceInWei <= 0;
+
         return {
           symbol: token.symbol,
           totalRequiredInWei,
           currentBalanceInWei,
+          areFundsReceived,
           decimals: token.decimals,
           isNative: token.tokenType === TokenType.NativeGas,
-        };
+        } satisfies TokenInfoProps;
       },
     );
   }, [bridgeRefillRequirements, masterEoa]);
 
+  // After the user has deposited the required funds,
+  // send the quote ID, cross-chain transfer details to the next step
+  useEffect(() => {
+    if (isRequestingQuote) return;
+    if (!bridgeRefillRequirements) return;
+    if (tokens.length === 0) return;
+
+    const areAllFundsReceived =
+      tokens.every((token) => token.areFundsReceived) &&
+      !bridgeRefillRequirements.is_refill_required;
+    if (!areAllFundsReceived) return;
+
+    updateQuoteId(bridgeRefillRequirements.id);
+    updateCrossChainTransferDetails({
+      fromChain: upperFirst(MiddlewareChain.ETHEREUM),
+      toChain: upperFirst(toMiddlewareChain),
+      transfers: tokens.map((token) => ({
+        fromSymbol: token.symbol,
+        fromAmount: token.currentBalanceInWei.toString(),
+        toSymbol: token.symbol,
+        toAmount: token.totalRequiredInWei.toString(),
```

Comment:
I'm thinking that this should be not `totalRequiredInWei` as it more represents how much funds are needed on the source chain for bridging (will be more than we receive in the end), but the requirements from the usual `refillRequirements`